theano/tensor/signal/tests/test_conv.py

```diff
@@ -82,14 +82,13 @@ def sym_conv2d(input, filters):
 
     def test_basic(self):
         """
-        Basic functionality of nnet.conv.ConvOp is already tested by
-        its own test suite.  We just have to test whether or not
-        signal.conv.conv2d can support inputs and filters of type
-        matrix or tensor3.
+        Basic functionality of nnet.abstract_conv.AbstractConvOp is
+        already tested by its own test suite.  We just have to test
+        whether or not signal.conv.conv2d can support inputs and
+        filters of type matrix or tensor3.
         """
-        if(not theano.tensor.nnet.conv.imported_scipy_signal and
-           theano.config.cxx == ""):
-            raise SkipTest("conv2d tests need SciPy or a c++ compiler")
+        if(theano.config.cxx == "" or theano.config.mode == "FAST_COMPILE"):
+            raise SkipTest("conv2d tests need a c++ compiler")
```

Comment:
I think the good fix would take 1 or 2 step:

- Add the tag "cxx_only" to all the cormm optimizations like:

conv_groupopt.register('local_abstractconv_gemm', local_abstractconv_gemm, 30,

- If there is still an error, try with this diff:

```
diff --git a/theano/compile/mode.py b/theano/compile/mode.py
index 1a34b6c..f97de81 100644
--- a/theano/compile/mode.py
+++ b/theano/compile/mode.py
@@ -56,6 +56,8 @@ OPT_FAST_RUN_STABLE = OPT_FAST_RUN.requiring('stable')
 # exclude them if we exclude 'gpu'.
 OPT_FAST_COMPILE = gof.Query(include=['fast_compile', 'fast_compile_gpu'],
                              exclude=exclude)
+OPT_FAST_COMPILE_NO_C = gof.Query(include=['fast_compile', 'fast_compile_gpu'],
+                                  exclude=['cxx_only'])
 OPT_STABILIZE = gof.Query(include=['fast_run'], exclude=exclude)
 OPT_STABILIZE.position_cutoff = 1.5000001
 OPT_NONE.name = 'OPT_NONE'
@@ -63,6 +65,7 @@ OPT_MERGE.name = 'OPT_MERGE'
 OPT_FAST_RUN.name = 'OPT_FAST_RUN'
 OPT_FAST_RUN_STABLE.name = 'OPT_FAST_RUN_STABLE'
 OPT_FAST_COMPILE.name = 'OPT_FAST_COMPILE'
+OPT_FAST_COMPILE.name = 'OPT_FAST_COMPILE_NO_C'
 OPT_STABILIZE.name = 'OPT_STABILIZE'
 
 predefined_optimizers = {
@@ -72,6 +75,7 @@ predefined_optimizers = {
     'fast_run': OPT_FAST_RUN,
     'fast_run_stable': OPT_FAST_RUN_STABLE,
     'fast_compile': OPT_FAST_COMPILE,
+    'fast_compile_no_c': OPT_FAST_COMPILE_NO_C,
     'stabilize': OPT_STABILIZE}
 
 
@@ -358,7 +362,7 @@ class Mode(object):
 # string as the key
 # Use VM_linker to allow lazy evaluation by default.
 FAST_COMPILE = Mode(theano.gof.vm.VM_Linker(use_cloop=False, c_thunks=False),
-                    'fast_compile')
+                    'fast_compile_no_c')
 if theano.config.cxx:
     FAST_RUN = Mode('cvm', 'fast_run')
 else:

                       'conv_gemm', 'fast_compile', 'fast_run', 'cxx_only')

```

With this, it should fall back to another implementation. Can you try that? thanks.